src/inc/til/env.h

```diff
@@ -0,0 +1,515 @@
+
+                    for (DWORD pass = 0; pass < 2; ++pass)
+                    {
+                        DWORD valueNameSize = maxValueNameSize;
+                        DWORD valueDataSize = maxValueDataSize;
+
+                        DWORD index = 0;
+                        DWORD type = 0;
+
+                        while (true)
+                        {
+                            DWORD result = RegEnumValueW(key.get(), index, valueName.data(), &valueNameSize, nullptr, &type, valueData.data(), &valueDataSize);
+                            if (result != ERROR_SUCCESS)
+                            {
+                                break;
+                            }
+                            valueName.resize(valueNameSize);
+                            valueData.resize(valueDataSize);
+
+                            if (valueNameSize)
+                            {
+                                std::wstring data;
+                                if (pass == 0 && (type == REG_SZ) && valueDataSize >= sizeof(wchar_t))
+                                {
+                                    data = {
+                                        reinterpret_cast<wchar_t*>(valueData.data()), valueData.size() / sizeof(wchar_t)
+                                    };
+                                }
+                                else if (pass == 1 && (type == REG_EXPAND_SZ) && valueDataSize >= sizeof(wchar_t))
+                                {
+                                    data = {
+                                        reinterpret_cast<wchar_t*>(valueData.data()), valueData.size() / sizeof(wchar_t)
+                                    };
+                                    data = expand_environment_strings(data.data());
+                                }
+
+                                // Because Registry data may or may not be null terminated... check if we've managed
+                                // to store an extra null in the wstring by telling it to create itself from pointer and size.
+                                // If we did, pull it off.
+                                if (!data.empty())
+                                {
+                                    if (data.back() == L'\0')
+                                    {
+                                        data = data.substr(0, data.size() - 1);
+                                    }
+                                }
+
+                                if (!data.empty())
+                                {
+                                    if (is_path_var(valueName))
+                                    {
+                                        concat_var(valueName, std::wstring{ data });
+                                    }
+                                    else
+                                    {
+                                        set_user_environment_var(valueName, std::wstring{ data });

```

Comment:
this will expand it even during pass 1 for a `REG_SZ`; is that definitely what we want? I suspect it is, because the environment is _strange_